guides/specification.md

```diff
@@ -0,0 +1,549 @@
+
+> Let's clarify with an example:
+>
+> A Check might define a fact named `awesome_fact` which is expected to be different given the _color_ of the execution.
+> - it has to be `cat` when the `color` in the execution context is `red`
+> - it has to be `dog` when the `color` in the execution context is `blue`
+> - it has to be `rabbit` in all other cases, regardless of the execution context
+> 
+> so we define a named variable `awesome_expectation` that resolves to `cat|dog|rabbit` when proper conditions are met
+>
+> allowing us to have an expectation like this 
+>
+> `expect: facts.awesome_fact == values.expected_token_timeout`
+
+In a nutshell, Values provide a way to decouple concepts, usually expeced values for facts, from their actual concretization during execution in certain scenarios.
+
+A Value declaration contains:
+- the value name
+- the default value
+- a list of conditions that determine the value given the context (optional, see [constant values](#constant-values))
+
+```yaml
+values:
+  - name: <value_name>
+    default: <default_value>
+    conditions:
+      - value: <value_on_condition_a>
+        when: <expression_a>
+      - value: <value_on_condition_b>
+        when: <expression_b>
+```
+
+It could read as: 
+
+the value named `<value_name>` resolves to 
+- `<value_on_condition_a>` when `<expression_a>` is true
+- `<value_on_condition_b>` when `<expression_b>` is true
+- `<default_value>` in all other cases
+
+Example:
+
+> Check `156F64 Corosync token timeout is set to expected value` defines a fact `corosync_token_timeout` which is expected to be different given the platform (aws/azure/gcp), so we define a named variable `expected_token_timeout` resolving to the appropriate value.
+>
+> `expected_token_timeout` resolves to:
+> - `30000` when `azure`/`aws` are detected
+> - `20000` on `gcp`
+> - `5000` in all other cases (ie: bare metal, VMs...)
+
+```yaml
+values:
+  - name: expected_token_timeout
+    default: 5000
+    conditions:
+      - value: 30000
+        when: env.provider == "azure" || env.provider == "aws"
+      - value: 20000
+        when: env.provider == "gcp"
+
+expectations:
+  - name: corosync_token_timeout_is_correct
+    expect: facts.corosync_token_timeout == values.expected_token_timeout
+```
+
+Note that `conditions` is a cascading chain of contextual inspection to determine which is the resolved value.
+
+- there may be many conditions
+- first condition that passes determines the value, following are not evaluated
+- `when` entry [Expression](#expression-language) has [access](#evaluation-scope) to gathered [facts](#facts-1) and [env](#env) evaluation scopes
+
+All the _resolved_ declared values would be registered in the [`values`](#values-1) namespaced evaluation scope.
+
+## Expectations
+
+Expectations section is where the assumptions about the state of a target infrastructure for a given scenario are formalized in expressions determining whether a check passes or not.
```

Comment:
Could we rephrase this in something like
"Expectations are assertions on the state of a target infrastructure. By using fact and values they are able to determine if a check passes or not."